dm/pkg/checker/table_structure.go

```diff
@@ -296,88 +361,183 @@ func (c *ShardingTablesChecker) Check(ctx context.Context) *Result {
 		Name:  c.Name(),
 		Desc:  "check consistency of sharding table structures",
 		State: StateSuccess,
-		Extra: fmt.Sprintf("sharding %s", c.name),
+		Extra: fmt.Sprintf("sharding %s,", c.targetTableID),
 	}
 
+	startTime := time.Now()
+	log.L().Logger.Info("start to check sharding tables")
 	var (
 		stmtNode      *ast.CreateTableStmt
-		firstTable    string
+		firstTable    *filter.Table
 		firstInstance string
+		inCh          = make(chan *checkItem, maxThreadNum)
+		checkWg       sync.WaitGroup
+		reMu          sync.Mutex
 	)
 
-	for instance, schemas := range c.tables {
-		db, ok := c.dbs[instance]
-		if !ok {
-			markCheckError(r, errors.NotFoundf("client for instance %s", instance))
-			return r
-		}
+	for instance, tables := range c.tableMap {
+		firstInstance = instance
+		firstTable = tables[0]
+		break
+	}
+	db, ok := c.dbs[firstInstance]
+	if !ok {
+		markCheckError(r, errors.NotFoundf("client for instance %s", firstInstance))
+		return r
+	}
 
-		parser2, err := dbutil.GetParserForDB(ctx, db)
+	parser2, err := dbutil.GetParserForDB(ctx, db)
+	if err != nil {
+		r.Extra = fmt.Sprintf("fail to get parser for instance %s on sharding %s", firstInstance, c.targetTableID)
+		markCheckError(r, err)
+		return r
+	}
+	r.Extra = fmt.Sprintf("instance %s on sharding %s", firstInstance, c.targetTableID)
+	statement, err := dbutil.GetCreateTableSQL(ctx, db, firstTable.Schema, firstTable.Name)
+	if err != nil {
+		markCheckError(r, err)
+		return r
+	}
+
+	stmt, err := parser2.ParseOneStmt(statement, "", "")
+	if err != nil {
+		markCheckError(r, errors.Annotatef(err, "statement %s", statement))
+		return r
+	}
+
+	stmtNode, ok = stmt.(*ast.CreateTableStmt)
```

Comment:
how about wrap get stmtNode code (L396~L406) to a func? this can resued in  `checkFunc`